paasta_tools/utils.py

```diff
@@ -2434,6 +2444,31 @@ def get_ldap_reader_password(self) -> str:
     def get_default_push_groups(self) -> List:
         return self.config_dict.get("default_push_groups", None)
 
+    def get_git_config(self) -> Dict:
+        """Gets git configuration. Includes repo names and their git servers.
+
+        :returns: the git config dict
+        """
+        return self.config_dict.get(
+            "git_config",
+            {
+                "git_user": "git",
+                "repos": {
+                    "yelpsoa-configs": {
+                        "repo_name": "yelpsoa-configs",
+                        "git_server": DEFAULT_SOA_CONFIGS_GIT_URL,
+                    },
+                },
+            },
+        )
+
+    def get_git_repo_config(self, repo_name: str) -> Dict:
+        """Gets the git configuration for a specific repo.
+
+        :returns: the git config dict for a specific repo.
+        """
+        return self.get_git_config().get("repos", {}).get(repo_name, {})
```

Comment:
this is maybe still a bit on the over complex side since I'm not sure we have plans to specify other repos here.